src/stage1/api_calls.rs

```diff
@@ -57,9 +63,29 @@ pub(crate) fn get_os_versions(api_endpoint: &str, api_key: &str, device: &str) -
 
     let status = res.status();
     if status == 200 {
-        Ok(res
-            .json::<Versions>()
-            .upstream_with_context("Failed to parse request results")?)
+        // The API call returns a response with the following structure:
+        // {
+        //     "d": [
+        // {
+        //     "raw_version": "5.1.20+rev1"
+        // },
+        // {
+        //     "raw_version": "5.1.20"
+        // }
+        //     ]
+        // }
```

Comment:
Love that you added this! But the indentation doesn't seem right, at least when viewing on the GitHub UI. Suggestion:

```suggestion
        // {
        //     "d": [
        //         {
        //             "raw_version": "5.1.20+rev1"
        //         },
        //         {
        //             "raw_version": "5.1.20"
        //         }
        //     ]
        // }
```

Reply:
Fixed! good catch